Bottom: d7025eaf1efa36a16574a64f282dd4e4c9c708c5
Top:    9477a0eb746fd7494199a9bb4151d0b2d17cc961
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-02-15 19:47:54 +0000

Tmp changes


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 054e9bc..4351a78 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1703,16 +1703,19 @@ Collect (char poly, jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes
   VNODE *cur;
   DIRECTION dir = UNINITIALISED;
 
-  cur = (&a->head); //    ->next->next->next->next->next; /* Breaks circ_seg_test9.pcb */
+//  cur = (&a->head)->next->next->next->next->next; /* Breaks circ_seg_test9.pcb */
+//  cur = (&a->head); //    ->next->next->next->next->next; /* Breaks circ_seg_test9.pcb */
   cur = (&a->head)        ;
   do
     {
+#if 0
       // The following may be a nice speedup, but not sure if it is correct.
       // In particular, consider the case when we collect a 'B' polygon contour.
       // Could some of that countour may already have been collected, and there
       // still be a piece we are interested in after? (Can we reach it though??)
       if (cur->Flags.mark != 0)
         break;
+#endif
 
       if (j_rule (poly, cur, &dir) && cur->Flags.mark == 0)
 	Collect1 (e, cur, dir, contours, holes, j_rule);
